src/context.rs

```diff
@@ -78,6 +90,12 @@ impl ContextBuilder {
                 let path = std::ffi::CString::new(path)
                     .or(Err(Error::fatal(wolfssl_sys::WOLFSSL_BAD_PATH)))?;
                 if is_dir {
+                    // SAFETY: [`wolfSSL_CTX_load_verify_locations`][0] ([also][1]) is documented to get the valid `ctx` pointer from wolfSSL_CTX_new().
+                    // Either `file` or `path` can be valid path to a PEM file or a directory containing files.
```

Comment:
It seems odd to talk about "`file` or `path`" within the `is_dir` conditional (similarly in the `else` case), since we are in a context which is about a specific one of them.

Also "... can be valid path to a PEM file or a directory ..." is not quite pertinent here from a safety point of view, what's important is the validity of the pointer as a C-string -- if it happens to not actually point to a path/directory (or be an invalid path some other way) then that's not a safety issue -- i.e. WolfSSL won't crash or cause UB, it'll just return an error.

How about:

> If not NULL then the pointer passed as the `path` argument must be a valid NULL-terminated C-style string, which is guaranteed by the use of `std::ffi::CString::as_c_str()` here.

(and similar in the `else` case).